stdlib/websub/src/main/java/org/ballerinalang/net/websub/compiler/WebSubServiceCompilerPlugin.java

```diff
@@ -64,29 +81,30 @@ public void process(ServiceNode serviceNode, List<AnnotationAttachmentNode> anno
                                        + "' annotation");
         }
 
-        //        if (serviceType != null && GENERIC_SUBSCRIBER_SERVICE_TYPE.equals(serviceType.getTypeName()
-        // .getValue())) {
-        //            List<BLangResource> resources = (List<BLangResource>) serviceNode.getResources();
-        //            if (resources.size() > 2) {
-        //                dlog.logDiagnostic(Diagnostic.Kind.ERROR, serviceNode.getPosition(),
-        //                               "cannot have more than two resources with " + WEBSUB_PACKAGE + ":"
-        //                                       + GENERIC_SUBSCRIBER_SERVICE_TYPE);
-        //                return;
-        //            }
-        //            resources.forEach(res -> {
-        //                validateDefaultResources(res, dlog);
-        //            });
-        //
-        //            if (resources.size() < 1
-        //                    || (resources.size() == 1
-        //                                && !(RESOURCE_NAME_ON_NOTIFICATION.equals(resources.get(0).getName()
-        // .getValue())))) {
-        //                dlog.logDiagnostic(Diagnostic.Kind.ERROR, serviceNode.getPosition(),
-        //                               "required resource '" + RESOURCE_NAME_ON_NOTIFICATION + "' not "
-        //                                       + "specified with " +  WEBSUB_PACKAGE + ":" +
-        // GENERIC_SUBSCRIBER_SERVICE_TYPE);
-        //            }
-        //        }
+        BType listenerType = ((BLangService) serviceNode).listenerType;
+        if (listenerType != null && !WEBSUB_LISTENER.equals(listenerType.getDesc())) {
+            return;
+        }
+
+        List<BLangFunction> resources = (List<BLangFunction>) serviceNode.getResources();
+        if (resources.size() > 2) {
+            dlog.logDiagnostic(Diagnostic.Kind.ERROR, serviceNode.getPosition(),
+                               "cannot have more than two resources with " + WEBSUB_PACKAGE + ":"
+                                       + GENERIC_SUBSCRIBER_SERVICE_TYPE);
+            return;
+        }
+        resources.forEach(res -> {
+            validateDefaultResources(res, dlog);
+        });
+
+        if (resources.size() < 1
```

Comment:
Shouldn't this validation come before validateDefaultResources() ?

Reply:
fixed, thanks.